cmd/prometheus/main_test.go

```diff
@@ -357,6 +357,71 @@ func getCurrentGaugeValuesFor(t *testing.T, reg prometheus.Gatherer, metricNames
 	return res
 }
 
+func TestDeleteStorageDataOnStart(t *testing.T) {
+	for _, agentMode := range []bool{false, true} {
+		t.Run(fmt.Sprintf("%v", agentMode), func(t *testing.T) {
```

Comment:
question: Where is prometheus invoked here? How is it running and deleting these dirs?

Reply:
What do you mean? It's not, I only test the function which in invoked in `main`. The invocation is trivial, plus it's very difficult to test this by running Prometheus and validating e.g. old samples. We need proper e2e framework, at some point we could propose those to Prometheus (e.g. https://github.com/efficientgo/e2e)

Reply:
Once shipped I will manually test things with my setup, just in case.

Reply:
Oh - nvm. I was bleary-eyed when reviewing this and missed the `deleteStorageData` call on L366.